### PR TITLE
[SPARK-36186][PYTHON] Add as_ordered/as_unordered to CategoricalAccessor and CategoricalIndex

### PR DESCRIPTION
**Pull request by ueshin**
### What changes were proposed in this pull request?

Add `as_ordered`/`as_unordered` to `CategoricalAccessor` and `CategoricalIndex`.

### Why are the changes needed?

We should implement `as_ordered`/`as_unordered` in `CategoricalAccessor` and `CategoricalIndex` yet.

### Does this PR introduce _any_ user-facing change?

Yes, users will be able to use `as_ordered`/`as_unordered`.

### How was this patch tested?

Added some tests.

### CODE REVIEW

**Comment by dongjoon-hyun** [comment]
Hi, @maropu . It's irrelevant, but TPCDS GitHub Action job seems to fail on several PRs due to some reasons.

**Comment by maropu** [comment]
okay, I will look into the reason.

**Comment by dongjoon-hyun** [comment]
Thank you so much, @maropu !

**Comment by HyukjinKwon** [comment]
cc @xinrong-databricks and @itholic FYI

File: python/pyspark/pandas/indexes/category.py [hunk 1]
```diff
@@ -115,6 +115,10 @@ def __new__(cls, data=None, categories=None, ordered=None, dtype=None, copy=Fals
             )
         )
 
+    @property
+    def dtype(self) -> CategoricalDtype:
+        return cast(CategoricalDtype, super().dtype)
```
Comment (xinrong-databricks) [inline python/pyspark/pandas/indexes/category.py:120]: Nice!

**Review by xinrong-meng — APPROVED** [review]

**Comment by xinrong-databricks** [comment]
LGTM, thanks!

**Comment by ueshin** [comment]
@xinrong-databricks I updated a bit. Could you take another look? Thanks.

**Comment by xinrong-databricks** [comment]
The new commit looks fine. Thank you!

Test failures in CI jobs seem to be irrelevant.

**Review by HyukjinKwon — APPROVED** [review]

**Comment by ueshin** [comment]
The tests for pandas-on-Spark and the linter passed.
I'd merge this to master/3.2.

**Pull request closed without merging**